### PR TITLE
Fix comment error when nested

### LINKED ISSUE #81: 🐛 BUG: Error when formatting HTML comments
### Describe the Bug

Hi everyone, thanks for this plugin ! 

Using the basic index.astro from `yarn create astro`, this plugin returns an error when using the HTML comments : 
```
<!--
           - You can also use imported framework components directly in your markup!
           -
           - Note: by default, these components are NOT interactive on the client.
           - The `:visible` directive tells Astro to make it interactive.
           -
           - See https://docs.astro.build/core-concepts/component-hydration/  
 -->
```

I have this error in the prettier log : 
```
["ERROR" - 12:11:59] Error formatting document.
["ERROR" - 12:11:59] Cannot read property 'comments' of undefined
TypeError: Cannot read property 'comments' of undefined
	at addCommentHelper (g:\PROJECTS\Testing\ASTRO\astro_v0.21\node_modules\prettier\index.js:5972:25)
	at Object.addLeadingComment$3 [as addLeadingComment] (g:\PROJECTS\Testing\ASTRO\astro_v0.21\node_modules\prettier\index.js:5981:3)
	at attachCommentsHTML (g:\PROJECTS\Testing\ASTRO\astro_v0.21\node_modules\prettier-plugin-astro\dist\index.js:365:27)
	at Object.print (g:\PROJECTS\Testing\ASTRO\astro_v0.21\node_modules\prettier-plugin-astro\dist\index.js:501:9)
```

When you remove the comment, everything works fine.

Hope it helps ! Thanks !

### Steps to Reproduce

1. `npm create astro` using the basic template.
2. `yarn add --dev prettier-plugin-astro` to add the plugin.
3. Format the index.astro page in the src/pages folder
4. Error! The plugin doesn't work
5. Remove the comment
6. The plugin works

### PR DESCRIPTION
**Pull request by antonyfaris**
## Changes

Fixes #81 

## Testing

I added a test for nested comments.

## Docs

### CODE REVIEW

File: src/printer.ts [hunk 1]
```diff
@@ -128,7 +128,7 @@ function print(path: AstPath, opts: ParserOptions, print: printFn): Doc {
   }
 
   // 2. attach comments shallowly to children, if any (https://prettier.io/docs/en/plugins.html#manually-attaching-a-comment)
-  if (!isPreTagContent(path) && !isMarkdownSubDoc) {
+  if (!isPreTagContent(path) && !isMarkdownSubDoc && node.type === 'Fragment') {
```
Comment (antonyfaris) [inline src/printer.ts:131]: I think it should only run once at the top level.

File: src/utils.ts [hunk 2]
```diff
@@ -535,7 +535,11 @@ export function attachCommentsHTML(node: anyNode): void {
         nodesToRemove.push(next); // if arbitrary whitespace between comment and node, remove
         next++; // skip to the next non-whitespace node
       }
-      util.addLeadingComment(node.children[next], node.children[n]);
+      const commentNode = node.children[next];
+      const comment = node.children[n];
+      if (commentNode) {
```
Comment (antonyfaris) [inline src/utils.ts:540]: It checks if there is a node before trying to attach the comment.

**Review by jonathantneal — APPROVED** [review]
Seem like good improvements. The tests 🤌 make it easy to approve.

**Review by natemoo-re — APPROVED** [review]
Looks great, thanks!

File: src/utils.ts [hunk 3]
```diff
@@ -535,7 +535,11 @@ export function attachCommentsHTML(node: anyNode): void {
...
+        util.addLeadingComment(commentNode, comment);
```
Comment (boehs) [inline src/utils.ts:541]:
```suggestion
      const commentNode = node.children[next];
      if (commentNode) {
        util.addLeadingComment(commentNode, node.children[n]);
```

Comment is only used in one place, so it *may* make more sense to include it directly in the function. This is at the cost of clarity, so both ways work

Comment (boehs) [inline src/utils.ts:541]:
```suggestion
      const commentNode = node.children[next];
      if (commentNode) {
        const comment = node.children[n];
        util.addLeadingComment(commentNode, comment);
```

**Review by boehs — CHANGES_REQUESTED** [review]

**Pull request merged**